acoustic_msgs/msg/SonarImage.msg

```diff
@@ -1,29 +1,52 @@
-# Single scan from an imaging sonar
+# A single scan from an imaging sonar
 #
-# The X-axis is centered in the plane of the fan ("forward"), with Z down,
-# in order to be consistent with the NED convention.
+# Imaging sonars produce an array of sonar beams radiating from a 
+# common center.  In the most common configuration, the sonar produces
+# a fan shaped array of beams.
+# 
+# Data in this message must be in a North-East-Down (NED) coordinate frame.  
+# In the common fan-shaped array, the central axis of each beam lies on 
+# the X-Y axis, with each beam having an azimuth (rotation around
+# the Z-axis) which is given in the vector "azimuth_angles".
+# Zero azimuth is typically "straight ahead" from the physical sensor,
+# and azimuth angles follow the NED convention, with angles increasing 
+# to starboard.   Azimuth angles do not need to be evenly spaced.
 #
-# For sonars with a 1D array, the beams lie on the X-Y plane, and
-# and each beam has an azimuth (rotation about Z) value.  Zero azimuth
-# is in the direction of the X-axis, typically "straight out" from the sonar.
-# Elevation is the orthogonal direction (rotation about Y),
-# often termed the "vertical aperture"
+# Elevation is defined as rotation around the +Y axis, with negative
+# elevations "pitched down" and positive elevations "pitched up".  
+# For the majority of sonars, all beams are centered in elevation on the 
+# X-Y plane, such that all have an elevation of 0.  In this case
+# "elevation_angles" can be left empty. If the elevation
+# differs on a beam-by-beam basis, values can be provided in the 
+# "elevation_angles" field, however in that case
+#
+#   len(azimuth_angles) must equal len(elevation_angles)
+#
+# For each beam, the sonar collects a time sequence of acoustic returns.
+# We assume an estimate of speed of sound is available, even if only
+# a constant, to convert round trip travel times to ranges in meters,
+# which are quantized into a fixed number of range bins.   Distance from the
+# sensor to the center of each range bin is reported in the "ranges" vector.
+#
+# The measured acoustic return is stored for each range bin, for each 
+# azimuthal beam, in the "intensities[]" array, as an azimuth-major array.
+# That is, the array contains returns for every range in sequence for the
+# first azimuth value, then the returns for every range for the second azimuth
+# value, and so on.   The precise interpretation or units of the intensities 
+# will vary by sensor manufacturer.
 
-# The meaning of the header timestamp relative to the sonar's data
-# acquisition cycle is defined by the sonar driver. It is taken to be
-# a best estimate of the time when the sonar data is valid, which will
-# the timing information available from the sonar hardware.
+# The relationship between the header timestamp is the best 
```

Comment:
I'm still not thrilled with this -- though neither the old nor the new is grammatically correct =)

How about:
"""
The header timestamp gives the best estimate of the time for which the data is valid.

The exact relationship between timestamp and the sonar's data acquisition cycle is determined by the sonar driver.
"""